### PR TITLE
Makes role datums actually fucking work

### PR DESCRIPTION
**Pull request by MadmanMartian**
Things to fix

- [x] The isantag checks

- [x] Fixes gamemode start not creating and recruiting people to factions/roles

- [x] Can actually start a game and be assigned to a faction and it give you the objectives

- [x] Save the objectives to the users memory

- [ ] Reimplement the game actually ending

- [ ] Re implement lost antag stuff

- [ ] Fix the game panel where you can force a game mode

- [ ] Countless other things

### CODE REVIEW

File: code/controllers/configuration.dm [hunk 1]
```diff
@@ -183,22 +183,22 @@
 
 /datum/configuration/New()
 	. = ..()
-	var/list/L = typesof(/datum/game_mode) - /datum/game_mode
+	var/list/L = typesof(/datum/gamemode) - /datum/gamemode
```
Comment (ShiftyRail) [inline code/controllers/configuration.dm]: subtypesof(/datum/gamemode), if you care to

**Comment by MadmanMartian** [comment]
admin/topic.dm 1315

File: code/__HELPERS/_macros.dm [hunk 2]
```diff
@@ -157,31 +157,35 @@
 
 //Macros for antags
 
-#define isvampire(H) 1//((H.mind in ticker.mode.vampires) || H.mind.vampire)
+#define isvampire(H) H.mind && H.mind.GetRole(VAMPIRE)
 
-#define iscult(H) 1//(H.mind in ticker.mode.cult)
+#define iscult(H) H.mind && (iscultist(H) || isculthead(H) || faction == "cult")
```
Comment (ShiftyRail) [inline code/__HELPERS/_macros.dm:162]: The define CULT already exists, you may have forgot to apply it here.

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -1534,19 +1534,22 @@ Game Mode config tags:
...
-		else if(ticker.mode.name == "mixed")
-			var/datum/game_mode/mixed/mixed_mode = ticker.mode
-			for(var/datum/game_mode/GM in mixed_mode.modes)
-				if(GM.config_tag == mode_ctag)
-					found_mode = GM
-					break
 	return found_mode
 
+/proc/find_active_faction(var/faction_name)
+	var/found_faction = null
```
Comment (ShiftyRail) [inline code/__HELPERS/unsorted.dm:1545]: For clarity, you can declare this as a var/datum/faction, unless you want this as an id.

**Comment by MadmanMartian** [comment]
In order to test the game panel force-creating a gamemode, I need to remake each gamemode. Dis shud b gud

**Comment by DeityLink** [comment]
Will I still be able to force secret mixed rounds that start with half the crew being either cult/ling/tator/vamp/wizard ?

**Comment by SysadminJeroen** [comment]
Don't let this end into a meme that will never happen, like snowmap or a
complete SS13 remake that is not running in BYOND.

Op 22 jan. 2018 14:51 schreef "DeityLink" <notifications@github.com>:

**Comment by DeityLink** [comment]
I'm pretty sure it's happening for real. We're all rooting for it.

File: __DEFINES/role_datums_defines.dm [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Faction IDs
+
+#define CULT "cult"
+#define CULT_NARSIE "Cult of Nar-sie"
```
Comment (DeityLink) [inline __DEFINES/role_datums_defines.dm:4]: Nar-**S**ie

File: code/datums/gamemode/factions/cult.dm [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		word=pick(allwords)
+	var/wordexp = "[cultwords[word]] is [word]..."
+	to_chat(cult_mob, "<span class='sinister'>You remember one thing from the dark teachings of your master... [wordexp]</span>")
+	cult_mob.mind.store_memory("<B>You remember that</B> [wordexp]", 0, 0)
+
+
+/datum/faction/cult/machine
+	name = "Cult of Ratvar"
+	desc = "When engineers go just too far."
+	deity_name = "The exiled one"
```
Comment (DeityLink) [inline code/datums/gamemode/factions/cult.dm]: The **E**xiled **O**ne

Also add an empty line at the end of the file.

File: code/datums/gamemode/factions/cult.dm [hunk 6]
```diff
@@ -0,0 +1,54 @@
+/datum/faction/cult
+	ID = CULT
+	var/eldergod
+	var/deity_name
```
Comment (DeityLink) [inline code/datums/gamemode/factions/cult.dm:4]: so this var is called "deity_name" but it holds their title instead of their name?

Comment (MadmanMartian) [reply code/datums/gamemode/factions/cult.dm:4]: It can be either or any.

File: code/__HELPERS/_macros.dm (hunk 2 above)
Comment (MadmanMartian) [reply code/__HELPERS/_macros.dm:162]: mobs don't use the define for some reason

**Comment by MadmanMartian** [comment]
TIME TO BREAK THE RULES

**Comment by SysadminJeroen** [comment]
AUNTIE

**Comment by SysadminJeroen** [comment]
DATUMS

**Pull request merged**